src/Request/Request.php

```diff
@@ -421,21 +421,34 @@ public function request($path, array $options = []): RequestOperationResult
             $options
         );
         $body = $response->getBody()->getContents();
-        try {
-            $body = \json_decode(
-                $body,
-                false,
-                512,
-                JSON_THROW_ON_ERROR
-            );
-        } catch (\JsonException $jsonException) {
-            $this->logger->debug($jsonException->getMessage());
+        $statusCode = $response->getStatusCode();
+        $headers = $response->getHeaders();
+
+        // Don't attempt to decode JSON if the response is expected to have no body.
+        if (
+            !in_array($statusCode, [204, 304], true)
+            && (!isset($headers['Content-Length']) || $headers['Content-Length'][0] != '0')
+            && !empty($body)
+        ) {
+            try {
+                $body = \json_decode(
+                    $body,
+                    false,
+                    512,
+                    JSON_THROW_ON_ERROR
+                );
+            } catch (\JsonException $jsonException) {
+                $this->logger->debug($jsonException->getMessage());
```

Comment:
We should prepend a string in front of jsonException->getMessage() to make this line easier to find should it happen again in the future.

Reply:
100% agree, and `$body` should not overwrite `$body` on line 434.  we should preserve the body _before_ attempting json decode, and print what was in it if the json decode fails.